activitysim/abm/models/vehicle_type_choice.py

```diff
@@ -0,0 +1,444 @@
+
+    # add new attribute to logit choice vehicle types
+    choices['vehicle_type'] = choices['vehicle_type'] + '_' + prob_choices
+
+    return choices
+
+
+def annotate_vehicle_type_choice(model_settings, trace_label):
+    """
+    Contains the calls to optionally annotate the persons, households, or
+    vehicles tables.  Adds columns to the corresponding output tables.
+
+    Parameters
+    ----------
+    model_settings : dict
+    trace_label : str
+
+    Returns
+    -------
+    """
+    # - annotate households table
+    if model_settings.get('annotate_households'):
+        households = inject.get_table('households').to_frame()
+        expressions.assign_columns(
+            df=households,
+            model_settings=model_settings.get('annotate_households'),
+            trace_label=tracing.extend_trace_label(trace_label, 'annotate_households'))
+        pipeline.replace_table("households", households)
+
+    # - annotate persons table
+    if model_settings.get('annotate_persons'):
+        persons = inject.get_table('persons').to_frame()
+        expressions.assign_columns(
+            df=persons,
+            model_settings=model_settings.get('annotate_persons'),
+            trace_label=tracing.extend_trace_label(trace_label, 'annotate_persons'))
+        pipeline.replace_table("persons", persons)
+
+    # - annotate vehicles table
+    if model_settings.get('annotate_vehicles'):
+        vehicles = inject.get_table('vehicles').to_frame()
+        expressions.assign_columns(
+            df=vehicles,
+            model_settings=model_settings.get('annotate_vehicles'),
+            trace_label=tracing.extend_trace_label(trace_label, 'annotate_vehicles'))
+        pipeline.replace_table("vehicles", vehicles)
+
+
+def iterate_vehicle_type_choice(
+        vehicles_merged,
+        model_settings,
+        model_spec,
+        locals_dict,
+        estimator,
+        chunk_size,
+        trace_label):
+    """
+    Iterates through household vehicle numbers and selects a vehicle type of
+    the form {body_type}_{age}_{fuel_type}. The preprocessor is run for each
+    iteration on the entire chooser table, not just the one for the current
+    vehicle number.  This allows for computation of terms involving the presence
+    of other household vehicles.
+
+    Vehicle type data is read in according to the specification and joined to
+    the alternatives. It can optionally be included in the output vehicles table
+    by specifying the `COLS_TO_INCLUDE_IN_VEHICLE_TABLE` option in the model yaml.
+
+    Parameters
+    ----------
+    vehicles_merged : orca.DataFrameWrapper
+    model_settings : dict
+        yaml model settings file as dict
+    model_spec : pandas.DataFrame
+        omnibus spec file with expressions in index and one column per segment
+    locals_dict : dict
+    estimator : Estimator object
+    chunk_size : orca.injectable
+    trace_label : str
+
+    Returns
+    -------
+    all_choices : pandas.DataFrame
```

Comment:
i think it is returning a tuple of these? or is it either or?  please specify

Reply:
They are concatenated -- added additional documentation to make this clear.